src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -90,8 +119,13 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope],
               complete(StatusCodes.Created -> retryFailedDevices(ns, campaign, request))
             }
           } ~
-          (get & path("stats")) {
-            complete(campaigns.campaignStats(campaign.id))
+          get {
+            path("stats") {
+              complete(campaigns.campaignStats(campaign.id))
+            } ~
+            (path("failed-installations.csv") & parameter('failureCode.as[String])) {
```

Comment:
Should `csv` be in the path? Could be parameter in future?

Reply:
I remember we discussed this with @simao and we came to the conclusion that it was better in the path, although I can't recall why. I'll try to dig up that conversation. But yes, we could always change it to be in the query.

Reply:
IMO:

The restful way to request a format of a resource is the`Accept` header, however, many clients do not support this, for example some browsers, and it can be hard to use for some clients, so it's customary to use the file extension to request a specific format.

Reply:
We have a SPA, it should be able to manipulate headers...

Reply:
It wasn't last time we implemented this.

Reply:
Weird... What about adding the logic to Web App?